### PR TITLE
Use the correct version of tools (from go.mod)

### PR DESCRIPTION
**Pull request by asalkeld**
Currently if you have a different version of the tools in GO_BIN something will fail.
I happened to have a slightly different version of controller-gen and kept getting manifest diffs.
By using the version specified in go.mod we avoid this annoyance.

### CODE REVIEW

**Comment by dhellmann** [comment]
Nice!

/approve

File: Makefile [hunk 1]
```diff
@@ -18,10 +18,9 @@ COVER_PROFILE = cover.out
 BIN_DIR := bin
 
 CRD_OPTIONS ?= "crd:trivialVersions=false,allowDangerousTypes=true,crdVersions=v1"
-CONTROLLER_TOOLS_VERSION=v0.4.1
-CONTROLLER_GEN := $(BIN_DIR)/controller-gen
-GOLANGCI_LINT := $(BIN_DIR)/golangci-lint
-KUSTOMIZE := $(BIN_DIR)/kustomize
+CONTROLLER_GEN ?= go run sigs.k8s.io/controller-tools/cmd/controller-gen
+GOLANGCI_LINT ?= GOLANGCI_LINT_CACHE=$(GOLANGCI_LINT_CACHE) go run github.com/golangci/golangci-lint/cmd/golangci-lint
+KUSTOMIZE ?= go run sigs.k8s.io/kustomize/kustomize/v3
```
Comment (fmuyassarov) [inline Makefile:23]: Do we still need to keep [tools/install_kustomize.sh](https://github.com/metal3-io/baremetal-operator/blob/master/tools/install_kustomize.sh) ?

**Review by fmuyassarov — COMMENTED** [review]
Integration test failed when running [Install Ironic](https://github.com/metal3-io/metal3-dev-env/blob/40e3f79b78d4f317e117b98731230a7c929f5307/vm-setup/roles/v1aX_integration_test/tasks/move.yml#L65) ansible task from metal3-dev-env. The task is running [tools/deploy.sh](https://github.com/metal3-io/baremetal-operator/blob/master/tools/deploy.sh) script from BMO repo, which includes a [Makefile target ](https://github.com/metal3-io/baremetal-operator/blob/c74264a6ddcb8aa28e411bcbedb005173d770806/tools/deploy.sh#L205 ) to create Kustomize binary. That line can probably be removed and all the referenced kustomize commands in `deploy.sh` also need to be changed accordingly.
CI output:
```
TASK [v1aX_integration_test : Install Ironic] **********************************
[1;30mtask path: /home/****/metal3/vm-setup/roles/v1aX_integration_test/tasks/move.yml:65[0m
[0;31mfatal: [localhost]: FAILED! => {[0m
[0;31m    "changed": true,[0m
[0;31m    "cmd": "/home/****/tested_repo/tools/deploy.sh false true true true true",[0m
[0;31m    "delta": "0:00:00.405636",[0m
[0;31m    "end": "2021-03-12 11:10:09.423146",[0m
[0;31m    "invocation": {[0m
[0;31m        "module_args": {[0m
[0;31m            "_raw_params": "/home/****/tested_repo/tools/deploy.sh false true true true true",[0m
[0;31m            "_uses_shell": true,[0m
[0;31m            "argv": null,[0m
[0;31m            "chdir": null,[0m
[0;31m            "creates": null,[0m
[0;31m            "executable": null,[0m
[0;31m            "removes": null,[0m
[0;31m            "stdin": null,[0m
[0;31m            "stdin_add_newline": true,[0m
[0;31m            "strip_empty_ends": true,[0m
[0;31m            "warn": true[0m
[0;31m        }[0m
[0;31m    },[0m
[0;31m    "msg": "non-zero return code",[0m
[0;31m    "rc": 2,[0m
[0;31m    "start": "2021-03-12 11:10:09.017510",[0m
[0;31m    "stderr": "make[1]: *** No rule to make target 'bin/kustomize'.  Stop.",[0m
[0;31m    "stderr_lines": [[0m
[0;31m        "make[1]: *** No rule to make target 'bin/kustomize'.  Stop."[0m
[0;31m    ],[0m
[0;31m    "stdout": "~/tested_repo ~/metal3/vm-setup\nmake[1]: Entering directory '/home/****/tested_repo'\nmake[1]: Leaving directory '/home/****/tested_repo'",[0m
[0;31m    "stdout_lines": [[0m
[0;31m        "~/tested_repo ~/metal3/vm-setup",[0m
[0;31m        "make[1]: Entering directory '/home/****/tested_repo'",[0m
[0;31m        "make[1]: Leaving directory '/home/****/tested_repo'"[0m
[0;31m    ][0m
[0;31m}[0m
```

**Comment by asalkeld** [comment]
> Integration test failed when running [Install Ironic](https://github.com/metal3-io/metal3-dev-env/blob/40e3f79b78d4f317e117b98731230a7c929f5307/vm-setup/roles/v1aX_integration_test/tasks/move.yml#L65) ansible task from metal3-dev-env. The task is running [tools/deploy.sh](https://github.com/metal3-io/baremetal-operator/blob/master/tools/deploy.sh) script from BMO repo, which includes a [Makefile target ](https://github.com/metal3-io/baremetal-operator/blob/c74264a6ddcb8aa28e411bcbedb005173d770806/tools/deploy.sh#L205) to create Kustomize binary. That line can probably be removed and all the referenced kustomize commands in `deploy.sh` also need to be changed accordingly.

Thanks for the pointer! I'll fix that right now.

File: Makefile (hunk 1 above)
Comment (asalkeld) [reply Makefile:23]: no, i'll remove, thanks

**Comment by fmuyassarov** [comment]
Most probably the CI failure this time is not about the patch, but a common issue in the CI itself. The issue is being worked on right now, and update will be give soon. Related [slack chat](https://kubernetes.slack.com/archives/CHD49TLE7/p1615895171009200)

**Comment by furkatgofurov7** [comment]
> Most probably the CI failure this time is not about the patch, but a common issue in the CI itself. The issue is being worked on right now, and update will be given soon. Related [slack chat](https://kubernetes.slack.com/archives/CHD49TLE7/p1615895171009200)

Thanks, hopefully, should be fine now.

/test-integration

**Pull request merged**